paddle/fluid/operators/slice_op.cc

```diff
@@ -148,9 +148,14 @@ class SliceOp : public framework::OperatorWithKernel {
  protected:
   framework::OpKernelType GetExpectedKernelType(
       const framework::ExecutionContext &ctx) const override {
+    auto *in_var = ctx.InputVar("Input");
+    bool is_tensor = in_var->IsType<framework::LoDTensor>();
+    platform::Place place = ctx.device_context().GetPlace();
+    if (is_tensor) {
+      place = in_var->Get<framework::LoDTensor>().place();
```

Comment:
maybe we should check whether the tensor is initialized firstly, call `.place()` may triggered `holder_ should not be null error`

Reply:
Thx, done